pedsnetdcc/abstract_transform.py

```diff
@@ -0,0 +1,89 @@
+from abc import ABCMeta, abstractmethod
+
+
+class Transform(object):
+    """Abstract class to define an interface for transforming the CDM.
+
+    The PEDSNet DCC makes custom modifications to the PEDSNet Common
+    Data Model for performance and other reasons.
+
+    For practical reasons, the transformations are effected by
+    creating a temporary table via a `CREATE TABLE foo_tmp AS SELECT
+    ... FROM foo ...` statement, where the SELECT may add additional
+    columns to `foo` (and include any joins necessary to populating
+    the additional columns).
+
+    Subclasses of the Transform class help to create these SELECT
+    statements and are also able produce any required index creation
+    statements.
+    """
+    __metaclass__ = ABCMeta
+
+    @classmethod
+    def pre_transform(cls, conn):
```

Comment:
I think this should probably take a dburi and a schema instead of a connection object, which is [not thread safe](http://docs.sqlalchemy.org/en/latest/core/connections.html#sqlalchemy.engine.Connection). I know that having every database executing method create its own connection to the database introduces additional overhead, but I think it will also make parallelizing the methods a lot less bug-prone and I think we will gain more from parallelizing than we will lose from the connection overhead. And we can always go back and implement a higher level connection carefully. Does that make sense to you?